<!DOCTYPE html>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
    <meta http-equiv="content-type" content="text/html; charset=utf-8">
    <meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport">
    <meta name="keywords" content="hexo, autumn">
    <title>
        Life is Worth Living
    </title>
    <!-- favicon -->
    
    <link rel="icon" href="https://cdn.jsdelivr.net/gh/frontendsophie/hexo-theme-autumn@1.0.0/source/img/favicon.ico">
    
    <link rel="stylesheet" href="/css/style.css">

    <!-- highlight -->
    <link rel="stylesheet" href="https://cdn.bootcss.com/highlight.js/9.12.0/styles/github-gist.min.css">
    <script src="//cdn.bootcss.com/highlight.js/9.2.0/highlight.min.js"></script>
    <script>
        hljs.initHighlightingOnLoad()
    </script>
    <script src="https://cdn.bootcss.com/jquery/3.3.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/frontendsophie/hexo-infinite-scroll@1.0.0/dist/hexo-infinite-scroll.min.css">
    <script src="https://cdn.jsdelivr.net/gh/frontendsophie/hexo-infinite-scroll@1.0.0/dist/hexo-infinite-scroll.min.js"></script>
    <script>
        infiniteScroll()

        // for mobile menu
        $(function () {
            $('.social-button').click(function () {
                if ($('.social-links').hasClass('hide-links')) {
                    $('.social-links').removeClass('hide-links')
                } else {
                    $('.social-links').addClass('hide-links')
                }
            })
        })
    </script>
</head>

    <body style="background: url(https://cdn.jsdelivr.net/gh/frontendsophie/hexo-theme-autumn@1.0.0/source/img/button-bg.png) #f3f3f3">
        <div class="container">
            <header class="header">
    <h1 class="title">
        <a href="/" class="logo">
            Life is Worth Living
        </a>
    </h1>
    <h2 class="desc">
        
    </h2>

    <nav class="links">
        <button class="social-button">
            menu
        </button>
        <ul class="social-links hide-links">
            
                <li>
                    <a href="https://github.com/Birdlet">
                        Github
                    </a>
                </li>
                
        </ul>
    </nav>
</header>
                <main class="main">
                    <article class="post">
    
    
    
    <h2 class="post-title">
        Access Ensemble Genome Database to BiomaRt of R
    </h2>
    <ul class="post-date">
        <li>
            2017-08-16
        </li>
        <li>
            Birdlet
        </li>
    </ul>
    <div class="post-content">
        <p><strong>Why We Hate Cilantro</strong></p>
<p>I want to analysis SNP in coding region of olfactory receptor. In 2012, researchers studied SNP correlated with cilantro preference, and they found an SNP <strong>rs72921001</strong> influence people’s feeling of cilantro. They said rs72921001 is a frequently accured SNP (an A -&gt; C in DNA sequence) in <strong>OR6A2</strong>, an olfactory receptor.</p>
<p>Resonable, right? I immediately think this could cause an amino acid missense mutation. However, when I searched this SNP in <a href="http://www.ensembl.org/index.html" target="_blank" rel="noopener"><strong>Ensembl</strong></a>, I found it is located at upstream flanking region of OR10A2, and OR6A2 is another gene… Now I think maybe the name is changed during this time, and rs72921001 might influence the expression of OR10A2 in gene level… Umm.</p>
<p>Can I get more imformation of SNP in olfactory receptor? Let’s try by R!</p>
<a id="more"></a>
<h2 id="Ensembl-Database"><a href="#Ensembl-Database" class="headerlink" title="Ensembl Database"></a>Ensembl Database</h2><p>Talking about genome database, <a href="http://www.ensembl.org/index.html" target="_blank" rel="noopener"><strong>Ensembl</strong></a> is the one I prefered and used. I’ll copy the introdution in their main page…</p>
<blockquote>
<p>Ensembl is a genome browser for vertebrate genomes that supports research in comparative genomics, evolution, sequence variation and transcriptional regulation. Ensembl annotate genes, computes multiple alignments, predicts regulatory function and collects disease data. Ensembl tools include BLAST, BLAT, BioMart and the Variant Effect Predictor (VEP) for all supported species.</p>
</blockquote>
<p>How can I access large dataset form Ensembl? Officialy they provide ready-made tools <a href="http://www.ensembl.org/info/docs/tools/index.html" target="_blank" rel="noopener"><strong>here</strong></a>. BioMart here is the most strong tool here. Acctually, what I really want to use here are scripts, so I found this: <a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html" target="_blank" rel="noopener"><strong>biomaRt</strong></a>. It can access large dataset in BioMart database, including Ensembl, COSMIC, Uniprot, HGNC, Gramene, Wormbase and dbSNP mapped to Ensembl.</p>
<h2 id="Access-Ensembl-Database-by-biomaRt"><a href="#Access-Ensembl-Database-by-biomaRt" class="headerlink" title="Access Ensembl Database by biomaRt"></a>Access Ensembl Database by biomaRt</h2><p>Firstly, install it by <strong>Bioconductor</strong>:<br><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## try http:// if https:// URLs are not supported</span></span><br><span class="line"><span class="keyword">source</span>(<span class="string">"https://bioconductor.org/biocLite.R"</span>)</span><br><span class="line">biocLite(<span class="string">"biomaRt"</span>)</span><br></pre></td></tr></table></figure></p>
<p>Then import <strong><em>biomaRt</em></strong> and <strong><em>Biostrings</em></strong> (for sequence processing)<br><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(biomaRt)</span><br><span class="line"><span class="keyword">library</span>(Biostrings)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">setwd(<span class="string">"C:\\Users\\Birdlet\\Desktop\\OR\\OR-Ensembl"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Set Ensembl Server</span></span><br><span class="line">ensembl = useEnsembl(biomart = <span class="string">"snp"</span>, dataset = <span class="string">"hsapiens_snp"</span>, mirror = <span class="string">'asia'</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># FUN: Fetch OR SNP data</span></span><br><span class="line">fetch_OR &lt;- <span class="keyword">function</span>(OR, OR_name)&#123;</span><br><span class="line">    filename = paste(<span class="string">'data\\'</span>, OR_name, <span class="string">'.csv'</span>, sep = <span class="string">''</span>)</span><br><span class="line">    fetched = getBM(attributes = c(<span class="string">'ensembl_gene_stable_id'</span>, <span class="string">'allele'</span>, <span class="string">'refsnp_id'</span>,</span><br><span class="line">                                   <span class="string">'refsnp_source'</span>, <span class="string">'consequence_type_tv'</span>, <span class="string">'minor_allele_freq'</span>,</span><br><span class="line">                                   <span class="string">'sift_score'</span>, <span class="string">'polyphen_score'</span>,</span><br><span class="line">                                   <span class="string">'ensembl_peptide_allele'</span>, <span class="string">'translation_start'</span>),</span><br><span class="line">                    filters = c(<span class="string">'ensembl_gene'</span>, <span class="string">'minor_allele_freq_second'</span>, <span class="string">'so_mini_parent_name'</span>),</span><br><span class="line">                    values = list(ensembl_gene = OR, freq = <span class="string">'0.001'</span>, </span><br><span class="line">                                  parent_name = <span class="string">'missense_variant'</span>),</span><br><span class="line">                    mart = ensembl)</span><br><span class="line">    write.table(fetched, file = filename, quote = <span class="literal">FALSE</span>, row.names = <span class="literal">FALSE</span>, </span><br><span class="line">              col.names = <span class="literal">FALSE</span>, sep = <span class="string">','</span>, fileEncoding = <span class="string">'utf-8'</span>)</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure></p>
<p>Firstly, I use <code>useEnsembl</code> function to initialize the web server, and I give paprameters to choose “SNP” of “Homo Sapiens”. For accelerating connection, I also use mirror in Asia rater than default.</p>
<p>Next, I defined a funtion for later. <code>getBM</code> funtion fetchs data from <strong>Ensembl</strong>, I want data including attributes of _’ensembl_gene_stable_id’, ‘allele’, ‘minor_allele_freq’_, and set _’minor_allele_freq_second’_ larger than <em>‘0.001’</em>. Then I can get genes’ SNP data with MAF larger than 0.1  %.</p>
<p>More details please see manual for <strong>biomRt</strong>. ;P</p>
<h2 id="Access-each-SNP-record-from-Ensembl"><a href="#Access-each-SNP-record-from-Ensembl" class="headerlink" title="Access each SNP record from Ensembl"></a>Access each SNP record from Ensembl</h2><p>After preperation, I can input a file of protein id, and wait for downloading data!</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">ENSG00000221858,G/A,rs9655672,dbSNP,missense_variant,<span class="number">0.00858626</span>,<span class="number">0.13</span>,<span class="number">0.361</span>,A/<span class="literal">T</span>,<span class="number">223</span></span><br><span class="line">ENSG00000221858,C/A/<span class="literal">T</span>,rs80128486,dbSNP,missense_variant,<span class="number">0.0081869</span>,<span class="number">0.07</span>,<span class="number">0.1</span>,L/I,<span class="number">279</span></span><br><span class="line">ENSG00000221858,C/A/<span class="literal">T</span>,rs80128486,dbSNP,missense_variant,<span class="number">0.0081869</span>,<span class="number">0.21</span>,<span class="number">0.06</span>,L/<span class="literal">F</span>,<span class="number">279</span></span><br><span class="line">ENSG00000221858,G/A,rs142231743,dbSNP,missense_variant,<span class="number">0.00419329</span>,<span class="number">0.08</span>,<span class="number">0.376</span>,V/I,<span class="number">183</span></span><br><span class="line">ENSG00000221858,<span class="literal">T</span>/A,rs143394460,dbSNP,missense_variant,<span class="number">0.00159744</span>,<span class="number">0</span>,<span class="number">0.986</span>,I/N,<span class="number">91</span></span><br><span class="line">ENSG00000221858,G/A,rs34947817,dbSNP,missense_variant,<span class="number">0.114217</span>,<span class="number">0.3</span>,<span class="number">0.007</span>,S/N,<span class="number">26</span></span><br></pre></td></tr></table></figure>
<h2 id="Update-Later…-Updated"><a href="#Update-Later…-Updated" class="headerlink" title="Update Later… Updated!"></a>Update Later… Updated!</h2><p>You can found more detail about how to use it in <a href="https://bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/biomaRt.html" target="_blank" rel="noopener">https://bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/biomaRt.html</a>. Also, you can simply use help in R, it’s really convinient in RStdio.</p>
<p>BTW, I dislike cilantro but now I can accept it… Genes and environments together determine what we are!</p>

    </div>
</article>
                </main>
                <aside class="aside">
                    <section class="aside-section">
                        
    <h1>Categories</h1>

    

                    </section>
                    <section class="aside-section">
                        
    <h1>Archives</h1>

    <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/">2018</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2017/">2017</a></li></ul>


                    </section>
                    <section class="aside-section tag">
                        
    <h1>Tags</h1>

    <ul class="tag-list"><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioinformatics/">Bioinformatics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Cheminformatics/">Cheminformatics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Computational-Biology/">Computational Biology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ensembl/">Ensembl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Notes/">Notes</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Papers/">Papers</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biomaRt/">biomaRt</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cheminfomatics/">cheminfomatics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/computational-biology/">computational biology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/contact-map/">contact map</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mdtrj/">mdtrj</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/oddt/">oddt</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/python/">python</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rdkit/">rdkit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/svg/">svg</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/test/">test</a></li></ul>

                    </section>
                </aside>
        </div>
    </body>

</html>